### PR TITLE
Add support for symbol precision on the app

### PR DESCRIPTION
**Pull request by lucca65**
## What issue does this PR close
Closes N/A

## Changes Proposed ( a list of new changes introduced by this PR)

- Makes the app understand that symbols actually are a compound of two information, a `symbol_code` and `precision`
- Update parsers, selection sets, encoders, decoders and type helpers on the whole app
- Updates some displays of amount and balance on the whole app

## How to test ( a list of instructions on how to test this PR)

This is actually though to test. We use both `asset` and `symbol` to almost every interaction on the app, since we use `symbol` as the community ID. So in order to test it we need to at least create two communities, one with a symbol with precision of 0 `0,XXX` and another with some precision `8,YYYY`. This means the test need to be done for:

- [x] Making transfers
- [x] Creating community
- [x] Creating objectives
- [x] Creating actions
- [x] Claiming actions
- [x] Checking claims
- [x] Change community
- [x] Create sale
- [x] Buy from a shop offer
- [x] Creating invitations
- [x] Accepting invitations

I count on your help @RodrigoLeiteF and @amiskov!!

### CODE REVIEW

File: src/elm/Page/Community/ObjectiveEditor.elm [hunk 1]
```diff
@@ -327,7 +329,7 @@ update msg model loggedIn =
                     UR.init model
                         |> updateObjective msg (\o -> { o | save = Saving })
 
-                save form isEdit =
+                save form isEdit precision =
```
Comment (amiskov) [inline src/elm/Page/Community/ObjectiveEditor.elm:332]: `precision` parameter is never used in this function.

**Review by amiskov — CHANGES_REQUESTED** [review]
- [ ] On the Shop pages (product list, product view) the precision digit looks odd:
image
I think we should show the only Symbol without the precision here: `10 MIZU`.
- [ ] Same for the list of transfers on the dashboard:
image
- [ ] After buying the product from the Shop got this error. It didn't affect the purchase, just seems like we need to do something with it:
image

I'll continue testing a bit later and will create another message if I'll find something.

**Comment by amiskov** [comment]
- [ ] Can't join the community with 2-digits precision. _For 0-precision community it worked fine_. I tried this link for the community I created today: `http://localhost:3000/invite/VrpQP5`. I can't join this community under an already registered user as well as I can't join this community creating a new user. Got this error: 
image

**Comment by amiskov** [comment]
@lucca65 I tested your list on 2 new communities: `0,ATMA` and `2,ATMB`. The first one with 0-precision worked well, but I stuck on the invitation step for the second one with 2-precision (I described it on the comment above).

**Comment by lucca65** [comment]
Nice! Thanks a lot @amiskov!!!! I'll debug right now!

File: src/index.js [hunk 2]
```diff
@@ -863,7 +863,11 @@ async function handleJavascriptPort (arg) {
       break
     }
     case 'setSelectedCommunity': {
-      devLog('=======================', 'storeSelectedCommunity')
+      devLog(
+        '=======================',
+        'storeSelectedCommunity',
+        arg.data.selectedCommunity
```
[the remark this answers is not in the document]
Comment (lucca65) [reply src/index.js:869]: Oops, wrong debugging artifact! 😀 Thanks !!!

**Comment by lucca65** [comment]
@amiskov the backend problem was fixed and deployed

**Comment by amiskov** [comment]
## Actions
- [ ] If the community has the precision of `0`, then the placeholder shouldn't contain any precision digits. E.g. this placeholder is misleading for the community with 0-precision (create Action page):
image
## Shop
- [ ] If the community has non-zero precision, then we should allow users to use fractional prices for their sells. Now we can use only whole numbers on "Create offer" page (`http://localhost:3000/shop/new/sell`):
image

**Comment by lucca65** [comment]
@amiskov just opened an issue #381 for the "After buying the product from the Shop got this error. It didn't affect the purchase, just seems like we need to do something with it:" comment mention

**Comment by lucca65** [comment]
@amiskov about the placeholders, I think we can create a component to improve that. Check #380, I've updated its description to include the fields you mentioned

**Comment by lucca65** [comment]
about filling the necessary 0s I've created this issue #383 (I would love to work on that!). An Eos library might come out of it!

**Review by amiskov — APPROVED** [review]

**Pull request merged**